nikola/post.py

```diff
@@ -235,7 +235,11 @@ def __repr__(self):
         m = hashlib.md5()
         # source_path modification date (to avoid reading it)
         m.update(utils.unicode_str(os.stat(self.source_path).st_mtime).encode('utf-8'))
-        m.update(utils.unicode_str(json.dumps(self.meta, cls=utils.CustomEncoder, sort_keys=True)).encode('utf-8'))
+        clean_meta = {}
+        for k, v in self.meta.items():
+            if v:
+                clean_meta[k] = v
```

Comment:
This unfortunately doesn't work: the problem is that meta is a dict from language to another dict, and the inner dict is the problem. I'll fix that.